ddtrace/tracer.py

```diff
@@ -211,6 +213,9 @@ def __init__(
         :param url: The Datadog agent URL.
         :param dogstatsd_url: The DogStatsD URL.
         """
+
+        maybe_start_serverless_mini_agent()
```

Comment:
We should avoid starting the mini agent if it can't be used to submit traces. We can defer this to a future PR

Reply:
Thoughts? @thedavl